src/main/java/com/minecolonies/permissions/ColonyPermissionEventHandler.java

```diff
@@ -0,0 +1,231 @@
+    {
+        if (checkBlockEventDenied(event.getWorld(), event.getPos(), event.getPlayer(), event.getPlacedBlock()))
+        {
+            cancelEvent(event);
+        }
+    }
+
+    private static void cancelEvent(final Event event)
+    {
+        event.setResult(Event.Result.DENY);
+        if (event.isCancelable())
+        {
+            event.setCanceled(true);
+        }
+    }
+
+    /**
+     * BlockEvent.BreakEvent handler.
+     *
+     * @param event BlockEvent.BreakEvent
+     */
+    @SubscribeEvent
+    public void on(final BlockEvent.BreakEvent event)
+    {
+        if (checkBlockEventDenied(event.getWorld(), event.getPos(), event.getPlayer(), event.getWorld().getBlockState(event.getPos())))
+        {
+            cancelEvent(event);
+        }
+    }
+
+    /**
+     * ExplosionEvent.Detonate handler.
+     *
+     * @param event ExplosionEvent.Detonate
+     */
+    @SubscribeEvent
+    public void on(final ExplosionEvent.Detonate event)
+    {
+
+        final World eventWorld = event.getWorld();
+        final Predicate<BlockPos> getBlocksInColony = pos -> colony.isCoordInColony(eventWorld, pos);
+        final Predicate<Entity> getEntitiesInColony = entity -> colony.isCoordInColony(entity.getEntityWorld(), entity.getPosition());
+        // if block is in colony -> remove from list
+        final List<BlockPos> blocksToRemove = event.getAffectedBlocks().stream()
+                                                .filter(getBlocksInColony)
+                                                .collect(Collectors.toList());
+
+        // if entity is in colony -> remove from list
+        final List<Entity> entitiesToRemove = event.getAffectedEntities().stream()
+                                                .filter(getEntitiesInColony)
+                                                .collect(Collectors.toList());
+        event.getAffectedBlocks().removeAll(blocksToRemove);
+        event.getAffectedEntities().removeAll(entitiesToRemove);
+    }
+
+    /**
+     * ExplosionEvent.Start handler.
+     *
+     * @param event ExplosionEvent.Detonate
+     */
+    @SubscribeEvent
+    public void on(ExplosionEvent.Start event)
+    {
+        if (colony.isCoordInColony(event.getWorld(), new BlockPos(event.getExplosion().getPosition())))
+        {
+            cancelEvent(event);
+        }
+    }
+
+    /**
+     * PlayerInteractEvent handler.
+     * <p>
+     * Check, if a player right clicked a block.
+     * Deny if:
+     * - If the block is in colony
+     * - block is AbstractBlockHut
+     * - player has not permission
+     *
+     * @param event PlayerInteractEvent
+     */
+    @SubscribeEvent
+    public void on(final PlayerInteractEvent event)
+    {
+        if (colony.isCoordInColony(event.getWorld(), event.getPos()))
+        {
+            final Block block = event.getWorld().getBlockState(event.getPos()).getBlock();
+            // Huts
+            if (block instanceof AbstractBlockHut
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) '&&' should be on a new line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AOperator_Wrap)